### PR TITLE
Split LinkageCheckerMainIntegrationTest.testBom by Java 8 and Java 11

### LINKED ISSUE #1827: LinkageCheckerMainIntegrationTest.testBom fails in Java 11
```
$ mvn test -Dtest=LinkageCheckerMainIntegrationTest#testBom
...
[ERROR] Tests run: 5, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 27.32 s <<< FAILURE! - in com.google.cloud.tools.opensource.classpath.LinkageCheckerMainIntegrationTest
[ERROR] testBom(com.google.cloud.tools.opensource.classpath.LinkageCheckerMainIntegrationTest)  Time elapsed: 21.828 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<Found 8[01] linkage errors> but was:<Found 8[23] linkage errors>
	at com.google.cloud.tools.opensource.classpath.LinkageCheckerMainIntegrationTest.testBom(LinkageCheckerMainIntegrationTest.java:134)
```

Comparing the dashboards build by Java 8 and Java 11 locally.

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #1827.

### CODE REVIEW

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java [hunk 1]
```diff
@@ -125,8 +127,11 @@ public void testArtifacts_noError()
   }
 
   @Test
-  public void testBom()
+  public void testBom_java8()
       throws IOException, RepositoryException, TransformerException, XMLStreamException {
+    // The number of linkage errors differ between Java 8 and Java 11 runtime.
+    Assume.assumeThat(System.getProperty("java.version"), StringStartsWith.startsWith("1.8.0"));
```
Comment (elharo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java:133]: 1.8.0 --> 1.8. for more generality

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java [hunk 2]
```diff
@@ -152,6 +157,45 @@ public void testBom()
                 + "/ com.google.appengine:appengine-api-1.0-sdk:1.9.71 (provided)");
   }
 
+  @Test
+  public void testBom_java11()
+      throws IOException, RepositoryException, TransformerException, XMLStreamException {
+    // The number of linkage errors differ between Java 8 and Java 11 runtime.
+    Assume.assumeThat(System.getProperty("java.version"), StringStartsWith.startsWith("11."));
```
Comment (elharo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java:164]: maybe greater than or equal to 11?

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java (hunk 1 above)
Comment (suztomo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java]: Updated.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java (hunk 2 above)
Comment (suztomo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java]: Updated. Yes, I want this test fail when we try next LTS JDK changes something related to linkage errors.

**Review by suztomo — COMMENTED** [review]
@elharo PTAL.

**Review by elharo — APPROVED** [review]

**Pull request merged**